src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynRuleSetGenerator.cs

```diff
@@ -50,62 +44,58 @@ public RuleSet Generate(string language, IEnumerable<SonarRule> rules)
     {
         _ = language ?? throw new ArgumentNullException(nameof(language));
         _ = rules ?? throw new ArgumentNullException(nameof(rules));
-
-        var ruleSet = new RuleSet
+        return new RuleSet
         {
             Name = "Rules for SonarQube",
             Description = "This rule set was automatically generated from SonarQube",
-            ToolsVersion = "14.0"
+            ToolsVersion = "14.0",
+            Rules = rules
+            .GroupBy(x => PartialRepoKey(x, language))
+            .Where(x => x.Key is not null)
+            .Select(CreateRules)
+            .ToList()
```

Comment:
Indent this +4 further so we can see where are properties and here are their values